packages/components/src/components/TableComponent/TableComponent.stories.js

```diff
@@ -0,0 +1,156 @@
+      { key: 'date', header: text('header3', 'Date Modified') }
+    ];
+
+    return (
+      <TableComponent
+        title={text('Title', 'Resource Name')}
+        dataRows={[]}
+        dataHeaders={dataHeaders}
+        selectedNamespace={text('selectedNamespace', '*')}
+        loading={boolean('loading', false)}
+      />
+    );
+  })
+  .add('table with one row, Add button, no checkboxes', () => {
+    const dataRows = [
+      {
+        id: 'namespace1:resource-one',
+        name: text('Name', 'resource-one'),
+        namespace: text('Namespace', 'namespace1'),
+        date: text('Date', '100 years ago')
+      }
+    ];
+
+    return (
+      <TableComponent
+        title={text('Title', 'Resource Name')}
+        dataRows={dataRows}
+        dataHeaders={[
+          { key: 'name', header: 'Name' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'date', header: 'Date Modified' }
+        ]}
+        selectedNamespace="*"
+        loading={boolean('loading', false)}
+        handleNew={action('handleNew')}
+      />
+    );
+  })
+  .add('table with one row, Add & Delete button and checkboxes', () => {
+    const dataRows = [
+      {
+        id: 'namespace1:resource-one',
+        name: text('Name', 'resource-one'),
+        namespace: text('Namespace', 'namespace1'),
+        date: text('Date', '100 years ago')
+      }
+    ];
+
+    return (
+      <TableComponent
+        title={text('Title', 'Resource Name')}
+        dataRows={dataRows}
+        dataHeaders={[
+          { key: 'name', header: 'Name' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'date', header: 'Date Modified' }
+        ]}
+        selectedNamespace="*"
+        loading={boolean('loading', false)}
+        handleNew={action('handleNew')}
+        handleDelete={action('handleDelete')}
+      />
+    );
+  });
+
+  /*
+  HOW TO USE:
+  - Required Props:
+      - title: String name of the Resource being surfaced
+
+      - dataHeaders: Array containing each column (in form of object)
+        appearing in data table.
+        Example:
+          [
+            { key: 'name', header: 'Name' },
+            { key: 'namespace', header: 'Namespace' },
+            { key: 'date', header: 'Date Modified' }
+          ]
+
+      - dataRows: Array containing each row (in form of object)
+        appearing in data table.
+        *all keys in dataHeaders must be specified in each row object*
+          Example:
+            [
+              {
+                **id: 'namespace1:resource-one',
+                name: 'resource-one',
+                namespace: 'namespace1',
+                date: '16 minutes ago'
+              }
+            ]
+          ** Must be specified for React to know which rows have changed,
+            are added, or are removed. It is also used when trying to
+            figure out which row(s) are about to be deleted. Finally, must
```

Comment:
finally, the ID must follow the pattern*, got confused reading this the very first time